sources/packages/backend/libs/integrations/src/atbc-integration/atbc-integration-processing.service.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable } from "@nestjs/common";
+import { LoggerService, InjectLogger } from "@sims/utilities/logger";
+import * as os from "os";
+import { SystemUsersService } from "@sims/services/system-users";
+import {
+  StudentService,
+  ATBCService,
+  ATBCCreateClientPayload,
+  ATBCCreateClientResponse,
+  ATBCPDStatus,
+} from "../services";
+import { Student } from "@sims/sims-db";
+import { ProcessPDRequestSummary } from "./models/atbc-integration.model";
+
+@Injectable()
+export class ATBCIntegrationProcessingService {
+  constructor(
+    private readonly studentService: StudentService,
+    private readonly atbcService: ATBCService,
+    private readonly systemUsersService: SystemUsersService,
+  ) {}
+
+  /**
+   * Apply for PD status calling the ATBC endpoint.
+   * @param studentId student who applied for PD status.
+   * @returns ATBC response.
+   */
+  async applyForPDStatus(studentId: number): Promise<ATBCCreateClientResponse> {
+    // Student who requested to apply PD status.
+    const student = await this.studentService.getStudentById(studentId);
+    const payload: ATBCCreateClientPayload = {
+      SIN: student.sinValidation.sin,
+      firstName: student.user.firstName,
+      lastName: student.user.lastName,
+      email: student.user.email,
+      birthDate: new Date(student.birthDate),
+    };
+    // Call ATBC endpoint to apply for PD status.
+    const response = await this.atbcService.createClient(payload);
+    const auditUser = await this.systemUsersService.systemUser();
+
+    // TODO: The if condition needs to be enhanced to check the response code
+    // TODO: to confirm that API call was successful. At this moment there is no information
+    // TODO: on how the response codes could be.
+    // TODO: The response code must be used to validate bad request as well.
```

Comment:
Please add only one `TODO` for every task/issue. The reason is because Sonar will count every one as a different issue.